sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/models/AmqpAnnotatedMessage.java

```diff
@@ -59,12 +69,15 @@ public AmqpAnnotatedMessage(AmqpAnnotatedMessage message) {
         AmqpMessageBodyType bodyType = message.getBody().getBodyType();
         switch (bodyType) {
             case DATA:
-                amqpMessageBody = AmqpMessageBody.fromData(message.getBody().getFirstData());
+                final byte[] data = message.getBody().getFirstData();
+                amqpMessageBody = AmqpMessageBody.fromData(Arrays.copyOf(data, data.length));
```

Comment:
Why are you copying the byte[] here?

Reply:
I actually removed copy constructor from `AmqpAnnotatedMessage`  , since this was created for service bus needs. dotnet also do not have copy constructor. 
And moved this logic into ServiceBusMessage, I still do Arrays.copyOf because when use is copying a ServiceBusReceivedMessage into ServiceBusMessage , changing one bytes should not effect bytes in other object. Thus I need to use  `Arrays.copyOf`

Reply:
You should clarify the behavior with @JoshLove-msft to ensure we are not doing needless copies. Specifically, where, if anywhere, does .NET copy in their amqp or service bus library. I would guess it is less than we do.

Reply:
We do not copy the bytes - we just create a new instance of AmqpMessageBody wrapping the same bytes - https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusMessage.cs#L66-L72

Reply:
Okay I will  remove this copy from here.